api/utils/sshutils/ssh.go

```diff
@@ -73,6 +73,40 @@ func SSHClientConfig(sshCert, privKey []byte, caCerts [][]byte) (*ssh.ClientConf
 	}, nil
 }
 
+// SSHClientConfigPrincipalsAsUser returns an ssh.ClientConfig with SSH credentials from this
+// Key and HostKeyCallback matching SSH CAs in the Key. The certificates's first valid principal
+// will be used as the ClientConfig's User.
+func SSHClientConfigPrincipalsAsUser(sshCert, privKey []byte, caCerts [][]byte) (*ssh.ClientConfig, error) {
```

Comment:
nit: this name is pretty confusing.
how about `SSHClientConfigWithValidPrincipal`?

Reply:
I agree this was confusing. I made refactors and renames to clean it up.